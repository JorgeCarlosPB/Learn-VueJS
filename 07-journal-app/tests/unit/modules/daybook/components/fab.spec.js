
import {shallowMount} from '@vue/test-utils'
import Fab from '@/modules/daybook/components/MyFab'

describe('Pruebas en el FAB component', ()=>{


    test('debe de mostrar el ícono por defecto', ()=>{
        //fa-plus
        const wrapper = shallowMount (Fab)
        expect (wrapper.html()).toMatchSnapshot()
        const iTag = wrapper.find('i')

        expect(iTag.classes('fa-plus')).toBeTruthy()

    })

    test('debe de mostrar el ícono por argumento: fa-circle', ()=>{
        //fa-circle
        const wrapper = shallowMount (Fab,{
            props: {
                icon: 'fa-circle'
            }
        })
        const iTag = wrapper.find('i')

        expect(iTag.classes('fa-circle')).toBeTruthy()
        
    })

    test('debe de emitir el evneto on:click cuando se hace click', ()=>{
        //wrapper.emmited('on:click')
        const wrapper = shallowMount (Fab)

        wrapper.find('button').trigger('click')

        expect(wrapper.emitted('on:click')).toHaveLength(1)
        
    })
})