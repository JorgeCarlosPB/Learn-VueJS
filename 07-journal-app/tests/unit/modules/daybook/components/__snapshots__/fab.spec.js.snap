// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el FAB component debe de mostrar el Ã­cono por defecto 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
